tests/e2e/isbservicerollout_test.go

```diff
@@ -0,0 +1,129 @@
+
+		// Get the latest ISBServiceRollout again before updating
+		gotISBServiceRollout, err := dynamicClient.Resource(gvr).Namespace(namespace).Get(ctx, isbServiceRolloutName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Update the ISBServiceRollout.
+		err = unstructured.SetNestedField(gotISBServiceRollout.Object, newisbServiceRolloutName, "metadata", "labels", "customLabelKey")
+		Expect(err).NotTo(HaveOccurred())
+		_, err = dynamicClient.Resource(gvr).Namespace(namespace).Update(ctx, gotISBServiceRollout, metav1.UpdateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Verify the update.
+		updatedISBServiceRollout, err := dynamicClient.Resource(gvr).Namespace(namespace).Get(ctx, isbServiceRolloutName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		val, _, _ := unstructured.NestedString(updatedISBServiceRollout.Object, "metadata", "labels", "customLabelKey")
+		Expect(val).To(Equal(newisbServiceRolloutName))
+
+		By("Deleting the ISBServiceRollout")
+		err = deleteISBServiceRollout(ctx, namespace, isbServiceRolloutName)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Checking the ISBServiceRollout has been deleted")
+		Eventually(func() bool {
+			_, err := dynamicClient.Resource(gvr).Namespace(namespace).Get(ctx, isbServiceRolloutName, metav1.GetOptions{})
+			return errors.IsNotFound(err)
+		}, timeout, duration).Should(BeTrue())
+	})
+
+})
+
+func createISBServiceRolloutSpec(name, namespace string) *unstructured.Unstructured {
+	spec := numaflowv1.InterStepBufferServiceSpec{
+		Redis: &numaflowv1.RedisBufferService{},
+		JetStream: &numaflowv1.JetStreamBufferService{
+			Version: "latest",
+			Persistence: &numaflowv1.PersistenceStrategy{
+				VolumeSize: &numaflowv1.DefaultVolumeSize,
+			},
+		},
+	}
+
+	// Convert spec to rawExtension
+	rawSpec, err := json.Marshal(spec)
+	if err != nil {
+		println(fmt.Sprintf("error marshaling spec: %v", err))
+	}
+
+	ISBServiceRollout := &apiv1.ISBServiceRollout{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "numaplane.numaproj.io/v1alpha1",
+			Kind:       "ISBServiceRollout",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: apiv1.ISBServiceRolloutSpec{
+			InterStepBufferService: runtime.RawExtension{Raw: rawSpec},
+		},
+	}
+
+	unstructuredObj, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(ISBServiceRollout)
+	return &unstructured.Unstructured{Object: unstructuredObj}
+}
+
+func createISBServiceRollout(ctx context.Context, isbServiceRollout *unstructured.Unstructured) error {
+	gvr := getGVRForISBService()
+	_, err := dynamicClient.Resource(gvr).Namespace(isbServiceRollout.GetNamespace()).Create(ctx, isbServiceRollout, metav1.CreateOptions{})
+	return err
+}
+
+func deleteISBServiceRollout(ctx context.Context, namespace, name string) error {
+	gvr := getGVRForISBService()
+	return dynamicClient.Resource(gvr).Namespace(namespace).Delete(ctx, name, metav1.DeleteOptions{})
```

Comment:
In this case the gvr should be for the ISBServiceRollout

Reply:
Please look over correct gvr for other Rollouts too

Reply:
@juliev0 do you mean to rename `getGVRForISBService` to `getGVRForISBServiceRollout` or change the Resource field here `Resource: "isbservicerollouts",`?

Reply:
You need this:
```

func getGVRForISBServiceRollout() schema.GroupVersionResource {
	return schema.GroupVersionResource{
		Group:    "numaplane.numaproj.io",
		Version:  "v1alpha1",
		Resource: "isbservicerollouts",
	}
}
func getGVRForISBService() schema.GroupVersionResource {
	return schema.GroupVersionResource{
		Group:    "numaflow.numaproj.io",
		Version:  "v1alpha1",
		Resource: "interstepbufferservices",
	}
}
```
and then to use each one at the right time according to your [document](https://docs.google.com/document/d/11UraoKLXYPiwM9_BbjwixwW38g3Pg6X3qlf8kzb57y4/edit#heading=h.admorc86w3t2)